lib/compress/zstd_compress.c

```diff
@@ -1936,6 +1936,13 @@ ZSTD_reset_matchState(ZSTD_matchState_t* ms,
         /* Row match finder needs an additional table of hashes ("tags") */
         size_t const tagTableSize = hSize * sizeof(U16);
         ms->tagTable = (U16 *) ZSTD_cwksp_reserve_aligned_init_once(ws, tagTableSize);
+        /* We want to generate a new salt in case we reset a Cctx, but we always want to use
+         * 0 when we reset a Cdict */
+        if(forWho == ZSTD_resetTarget_CCtx) {
+            ms->hashSalt = ms->hashSalt * 6364136223846793005 + 1; /* based on MUSL rand */
```

Comment:
There is probably enough time here to make `hashSalt` evolution more chaotic, closer to a prng.